newdle/client/src/components/UserSearch.js

```diff
@@ -0,0 +1,136 @@
+import React, {useCallback, useState} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+import {Button, Container, Icon, Label, List, Modal, Segment} from 'semantic-ui-react';
+import UserSearchForm from './UserSearchForm';
+import UserSearchResults from './UserSearchResults';
+import UserAvatar from './UserAvatar';
+import {updateParticipantList, removeParticipant} from '../actions';
+import {getMeetingParticipants} from '../selectors';
+import client from '../client';
+
+import styles from './UserSearch.module.scss';
+
+async function searchUsers(data, setResults) {
+  const q = Object.keys(data)
+    .filter(k => data[k])
+    .map(k => data[k])
+    .join(' ');
+
+  if (!q.trim()) {
+    return;
+  }
+  const results = await client.searchUsers(q);
+  setResults(results);
+}
+
+export default function UserSearch() {
+  const dispatch = useDispatch();
+  const participants = useSelector(getMeetingParticipants);
+  const [userModalOpen, setUserModalOpen] = useState(false);
+  const [searchResults, setSearchResults] = useState(null);
+  const [stagedParticipants, setStagedParticipants] = useState([]);
+
+  const handleRemoveParticipant = participant => dispatch(removeParticipant(participant));
+
+  const handleModalClose = useCallback(() => {
+    setUserModalOpen(false);
+    setSearchResults(null);
+    setStagedParticipants([]);
+  }, [setUserModalOpen, setSearchResults]);
```

Comment:
Isn't `setStagedParticipants` missing here?

Reply:
Yeah, good catch; )

Reply:
Don't know why the linter didn't complain about this :P